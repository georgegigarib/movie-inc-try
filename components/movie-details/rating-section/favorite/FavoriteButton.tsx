import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

interface FavoriteButtonProps {
  isFavorite: boolean;
  onToggleFavorite: () => void;
}

const FavoriteButton: React.FC<FavoriteButtonProps> = ({ isFavorite, onToggleFavorite }) => {
  return (
    <TouchableOpacity style={styles.favoriteButton} onPress={onToggleFavorite}>
      <Icon name={isFavorite ? 'heart' : 'heart-o'} size={30} color="#ff0000" />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  favoriteButton: {
    paddingVertical: 20,
  },
});

export default FavoriteButton;
