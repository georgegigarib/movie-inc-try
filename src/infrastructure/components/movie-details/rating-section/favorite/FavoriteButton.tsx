import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

interface FavoriteButtonProps {
  isFavorite: boolean;
  onToggleFavorite: () => void;
}

const FavoriteButton: React.FC<FavoriteButtonProps> = ({ isFavorite, onToggleFavorite }) => {
  return (
    <TouchableOpacity
      style={styles.favoriteButton}
      onPress={onToggleFavorite}
      testID="favorite-button"
    >
      <Icon
        name={isFavorite ? 'heart' : 'heart-o'}
        size={30}
        color="#ff0000"
        testID="icon"
      />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  favoriteButton: {
    paddingVertical: 20,
  },
});

export default FavoriteButton;
