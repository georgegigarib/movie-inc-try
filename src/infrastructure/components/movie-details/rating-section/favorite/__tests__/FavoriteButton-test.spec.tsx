import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import FavoriteButton from '@/src/infrastructure/components/movie-details/rating-section/favorite/FavoriteButton';

describe('FavoriteButton', () => {
  it('renders heart icon when isFavorite is true', () => {
    const { getByTestId, toJSON } = render(
      <FavoriteButton isFavorite={true} onToggleFavorite={() => {}} />
    );
    const icon = getByTestId('icon');
    expect(icon.props.children).toBe('heart');
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders heart-o icon when isFavorite is false', () => {
    const { getByTestId, toJSON } = render(
      <FavoriteButton isFavorite={false} onToggleFavorite={() => {}} />
    );
    const icon = getByTestId('icon');
    expect(icon.props.children).toBe('heart-o');
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onToggleFavorite when button is pressed', () => {
    const mockOnToggleFavorite = jest.fn();
    const { getByTestId } = render(
      <FavoriteButton isFavorite={false} onToggleFavorite={mockOnToggleFavorite} />
    );
    const button = getByTestId('favorite-button');
    fireEvent.press(button);
    expect(mockOnToggleFavorite).toHaveBeenCalledTimes(1);
  });
});
