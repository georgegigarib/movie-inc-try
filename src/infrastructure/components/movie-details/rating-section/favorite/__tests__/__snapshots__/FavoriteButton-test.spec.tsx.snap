// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton renders heart icon when isFavorite is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "paddingVertical": 20,
    }
  }
  testID="favorite-button"
>
  <Text
    style={
      {
        "color": "#ff0000",
        "fontSize": 30,
      }
    }
    testID="icon"
  >
    heart
  </Text>
</View>
`;

exports[`FavoriteButton renders heart-o icon when isFavorite is false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "paddingVertical": 20,
    }
  }
  testID="favorite-button"
>
  <Text
    style={
      {
        "color": "#ff0000",
        "fontSize": 30,
      }
    }
    testID="icon"
  >
    heart-o
  </Text>
</View>
`;
